### PR TITLE
Oy2 20157 - Waiver Amendment Form

### PR DESCRIPTION
**Pull request by bflynn-collabralink**
Story: https://qmacbis.atlassian.net/browse/OY2-20157
Endpoint: See github-actions bot comment

### Details

Create Waiver Amendment form for package view.

### Changes

- Created WaiverAmendmentForm and modified config as necessary

### Implementation Notes

- I reworked alot of the error/warning message handling around the ids to work as object arrays

### Test Plan

1. Login as state submitter
2. Navigate to package dashboard and select new submission -> Waiver Action
3. Verify label for Waiver Amendment card matches AC
4. Click Waiver Amendment option and verify you are routed to the new Waiver Amendment form.
5. Verify page title and sections per ACs
6. Enter an existing unapproved (any status other than approved) waiver number into Existing Waiver Number to Amend field
7. Verify error message per AC
8. Enter an existing approved waiver into Existing Waiver Number to Amend field
9. Verify no errors
10. Enter an invalid format waiver number into the 1915(b) Waiver Amendment Number field
11. Verify error message per AC
12. Enter a valid format waiver number into the 1915(b) Waiver Amendment Number field
13. Verify no errors
14. Fill out remaining required fields (Waiver Authority and attachments)
15. Verify submit button becomes active
16. Click Submit
17. Verify you are navigated back to package dashboard on the waiver tab and the green success banner appears at top

### CODE REVIEW

**Review by kristin-at-theta — APPROVED** [review]
Looks great - appreciate the cleanups, too!

I made comment about the casing for the triage choice... but this matches the AC, so *shrug* -- just be prepared for it to be a thing at demo... 

Tests correctly, automated tests have expected failures (nice cleanup of that duplicate id, btw)

Approved!

File: services/ui-src/src/libs/triageChoices.js [hunk 1]
```diff
@@ -51,7 +51,7 @@ export const choicesFromRoute = {
       },
       {
         title: "Waiver Amendment",
-        description: "Amend an Existing Waiver",
+        description: "Amend an Existing 1915(b) Waiver",
```
[the remark this answers is not in the document]
Comment (bflynn-collabralink) [reply services/ui-src/src/libs/triageChoices.js:54]: Yeah thats definitely not consistent, each card is all over the place. I agree they should all probably be in normal casing.... and maybe even with a period at the end :)

File: services/ui-src/src/page/OneMACForm.test.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ import { createMemoryHistory } from "history";
 import { Router } from "react-router-dom";
 import { stateSubmitterInitialAuthState } from "../libs/testDataAppContext";
 
-import { ROUTES, ONEMAC_ROUTES } from "cmscommonlib";
+import { ROUTES, ONEMAC_ROUTES, initialWaiver } from "cmscommonlib";
```
[the remark this answers is not in the document]
Comment (bflynn-collabralink) [reply services/ui-src/src/page/OneMACForm.test.js:13]: I could not do that if you think its better. My thinking was my test was not testing the proper title and I just wanted to ensure the test always gets the correct field to enter into irrespective of title. I do see your point though as we probably should make the test based on needing proper field titles as well based on current ACs.

File: services/ui-src/src/page/OneMACForm.tsx [hunk 3]
```diff
@@ -119,32 +105,47 @@ const OneMACForm: React.FC<{ formConfig: OneMACFormConfig }> = ({
 
   const validateComponentId = useCallback(
     (componentId) => {
-      let errorMessage = "";
+      let errorMessages: Message[] = [];
```
[the remark this answers is not in the document]
Comment (bflynn-collabralink) [reply services/ui-src/src/page/OneMACForm.tsx:108]: YES! Funny enough i dug myself into a giant hole going to far and replacing all ONEMAC_STATUS and ONEMAC_TYPE with imported versions from the corresponding form config files......ran into some crazy circular dependencies in the js loader and it couldnt resolve it properly. Had to back track and wasted like 2-3hrs of work. :(

**Pull request merged**